### PR TITLE
GPIO interrpt pins added for ESP32

### PR DESCRIPTION
**Pull request by NitiKaur**
@dpgeorge PR for Issue #6581, where I have added the interrupt pins for ESP32.
Also, I want to participate in GSOD but can't seem to contact my mentors thru any IRC Channel or Micropython forum or thru mail. If you know about them, can you please suggest me an alernative

### CODE REVIEW

File: docs/esp32/quickref.rst [hunk 1]
```diff
@@ -538,3 +538,31 @@ the corresponding functions, or you can use the command-line client
 
 See the MicroPython forum for other community-supported alternatives
 to transfer files to an ESP32 board.
+
+Interrupts
+----------
+**Hardware Interrupts** - They are fired by hardware peripherals and sources.hey can be: Internal or External. An example of internal hardware interrupts maybe something like a hardware timer interrupt or WDT. An example of an external hardware interrupt is external GPIO pins interrupts
```
Comment (cgreening) [inline docs/esp32/quickref.rst:544]: Small typo - "sources.hey" should be "sources. They"

**Comment by dpgeorge** [comment]
> Also, I want to participate in GSOD but can't seem to contact my mentors thru any IRC Channel or Micropython forum or thru mail

@NitiKaur please contact @mattytrentini

**Review by cgreening — APPROVED** [review]

File: docs/esp32/quickref.rst (hunk 1 above)
Comment (cgreening) [reply docs/esp32/quickref.rst]: Looks good to me - I'm not a maintainer so can't merge things unfortunately.

Comment (NitiKaur) [reply docs/esp32/quickref.rst]: @dpgeorge Please see if it can be merged

**Pull request closed without merging**